src/components/FilterChip/FilterChip.tsx

```diff
@@ -14,39 +14,43 @@ const FilterChipVariants = cva(
     },
   }
 );
-
-export interface FilterChipProps
-  extends React.HTMLAttributes<HTMLDivElement>,
-    VariantProps<typeof FilterChipVariants> {
+type addView = {
+  addView: true;
+  value: string;
+};
+type filterValue = {
+  addView: false;
   filter: string;
   value: string;
-  addView: boolean;
   onRemove?: () => void;
-}
+};
+export type FilterChipProps = React.HTMLAttributes<HTMLDivElement> &
+  VariantProps<typeof FilterChipVariants> &
+  (addView | filterValue);
```

Comment:
If you leave in onRemove in line 32, you probably need to switch this around to something like:

```suggestion
export type FilterChipProps = React.HTMLAttributes<HTMLDivElement> &
  VariantProps<typeof FilterChipVariants> &
  (addView | filterValue)  & {
    onRemove?: () => void; 
  };
```